### PR TITLE
Add optional support to security providers

### PR DESCRIPTION
**Pull request by arjav-desai**
Adding optional support to http-basic-auth, http-digest-auth and oidc-auth providers.

Also, fixed issue with jwt-auth provider throwing error/failure instead of abstain in some cases.

Addresses # 2772

### CODE REVIEW

File: security/providers/http-auth/src/main/java/io/helidon/security/providers/httpauth/HttpBasicAuthProvider.java [hunk 1]
```diff
@@ -263,11 +267,16 @@ private AuthenticationResponse invalidUser() {
...
         // DO NOT change this - it is a security problem if the message differs, as it gives too much information
         // to potential attacker
-        return fail("Invalid username or password");
+        return failOrAbstain("Invalid username or password");
     }
 
-    private AuthenticationResponse fail(String message) {
-        return AuthenticationResponse.builder()
+    private AuthenticationResponse failOrAbstain(String message) {
+        return optional ? AuthenticationResponse.builder()
```
Comment (tomas-langer) [inline security/providers/http-auth/src/main/java/io/helidon/security/providers/httpauth/HttpBasicAuthProvider.java:274]: Please do not use ternary expression for multiline choices. 
This is OK: `optional ? abstain(message) : fail(message)`
Or just use `if (optiona) {...`

File: security/providers/http-auth/src/main/java/io/helidon/security/providers/httpauth/HttpDigestAuthProvider.java [hunk 2]
```diff
@@ -200,19 +204,24 @@ private AuthenticationResponse validateDigestAuth(String headerValue, SecurityEn
...
 
-    private AuthenticationResponse fail(String message) {
-        return AuthenticationResponse.builder()
-                .statusCode(UNAUTHORIZED_STATUS_CODE)
-                .responseHeader(HEADER_AUTHENTICATION_REQUIRED, buildChallenge())
-                .status(AuthenticationResponse.SecurityStatus.FAILURE)
-                .description(message)
-                .build();
+    private AuthenticationResponse failOrAbstain(String message) {
+        return optional ? AuthenticationResponse.builder()
```
Comment (tomas-langer) [inline security/providers/http-auth/src/main/java/io/helidon/security/providers/httpauth/HttpDigestAuthProvider.java:214]: same as above

File: security/providers/jwt/src/main/java/io/helidon/security/providers/jwt/JwtProvider.java [hunk 3]
```diff
@@ -185,16 +174,28 @@ private AuthenticationResponse authenticateToken(String token) {
...
-                return AuthenticationResponse.failed(errors.toString());
+                return failOrAbstain(errors.toString());
             }
         } else {
             return AuthenticationResponse.success(buildSubject(signedJwt.getJwt(), signedJwt));
         }
     }
 
+    private AuthenticationResponse failOrAbstain(String message) {
+        return optional ? AuthenticationResponse.builder()
```
Comment (tomas-langer) [inline security/providers/jwt/src/main/java/io/helidon/security/providers/jwt/JwtProvider.java:188]: same as above

File: security/providers/oidc/src/main/java/io/helidon/security/providers/oidc/OidcProvider.java [hunk 4]
```diff
@@ -364,7 +366,25 @@ private AuthenticationResponse errorResponse(ProviderRequest providerRequest,
         }
     }
 
+    private AuthenticationResponse failOrAbstain(String message) {
+        return optional ? AuthenticationResponse.builder()
```
Comment (tomas-langer) [inline security/providers/oidc/src/main/java/io/helidon/security/providers/oidc/OidcProvider.java:370]: same as above

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: security/providers/http-auth/src/main/java/io/helidon/security/providers/httpauth/HttpBasicAuthProvider.java (hunk 1 above)
Comment (arjav-desai) [reply security/providers/http-auth/src/main/java/io/helidon/security/providers/httpauth/HttpBasicAuthProvider.java]: Replaced ternary  with if/else

File: security/providers/jwt/src/main/java/io/helidon/security/providers/jwt/JwtProvider.java (hunk 3 above)
Comment (arjav-desai) [reply security/providers/jwt/src/main/java/io/helidon/security/providers/jwt/JwtProvider.java]: Replaced ternary  with if/else

File: security/providers/oidc/src/main/java/io/helidon/security/providers/oidc/OidcProvider.java (hunk 4 above)
Comment (arjav-desai) [reply security/providers/oidc/src/main/java/io/helidon/security/providers/oidc/OidcProvider.java]: Replaced ternary  with if/else

File: security/providers/http-auth/src/main/java/io/helidon/security/providers/httpauth/HttpDigestAuthProvider.java (hunk 2 above)
Comment (arjav-desai) [reply security/providers/http-auth/src/main/java/io/helidon/security/providers/httpauth/HttpDigestAuthProvider.java]: Replaced ternary  with if/else

**Review by tomas-langer — APPROVED** [review]
LGTM

**Pull request merged**